client/tests/test.ts

```diff
@@ -1,6 +1,168 @@
-import { expect, test } from "@playwright/test";
+import { expect, test, type Frame, type Locator, type Page } from "@playwright/test";
 
-test("index page has expected h1", async ({ page }) => {
-	await page.goto("/");
-	await expect(page.getByRole("heading", { name: "Welcome to SvelteKit" })).toBeVisible();
+const TEST_URL = "https://example.com/test";
+
+type FormSubmit = {
+	address: string;
+	recaptcha: string;
+	parachain_id?: string;
+};
+
+const getFormElements = async (page: Page, getCaptcha = false) => {
+	let captcha: Locator = {} as Locator;
+	if (getCaptcha) {
+		// ?: Hack. We need to wait for the frame to load and then invade it.
+		await page.reload();
+		const captchaFrame = await new Promise<Frame>(function (resolve, reject) {
+			let i = 0;
+			// function that waits for the frame and timeouts after 3 seconds
+			(function waitForFrame() {
+				const captchaFrame = page
+					.frames()
+					.filter((f) => f.url().includes("https://www.google.com/recaptcha/api2/"));
+				if (captchaFrame.length > 0) {
+					return resolve(captchaFrame[0]);
+				} else {
+					i++;
+					if (i > 10) {
+						reject(new Error("Timout"));
+					}
+				}
+				setTimeout(waitForFrame, 300);
+			})();
+		});
+		captcha = captchaFrame?.locator("#recaptcha-anchor") as Locator;
+	}
+	return {
+		address: page.getByPlaceholder("Enter your address"),
+		useParachain: page.getByLabel("Use parachain"),
+		// value changes after useParachain is checked
+		parachain: page.getByPlaceholder("Using Relay chain"),
+		captcha,
+		submit: page.getByRole("button", { name: "Submit" })
```

Comment:
I'd recommend adding special markers (classname, data-attr, make your pick) to the elements in code, to search for them later in tests. 

"Element not found" errors are the nastiest.

Reply:
I second that.

Reply:
I had an issue while setting it up but now I fixed it. Thank you for your recommendation. Now the code is cleaner!